ouf.lua

```diff
@@ -564,6 +564,97 @@ function oUF:Spawn(unit, overrideName)
 	return object
 end
 
+function oUF:SpawnNamePlates(namePrefix, nameplateCallback, nameplateCVars)
+	argcheck(nameplateCallback, 3, 'function', 'nil')
+	argcheck(nameplateCVars, 4, 'table', 'nil')
+	if(not style) then return error('Unable to create frame. No styles have been registered.') end
+
+	local style = style
+	local prefix = namePrefix or generateName()
+
+	local hiddenParent = CreateFrame('Frame', nil, UIParent)
+	hiddenParent:SetAllPoints()
+	hiddenParent:Hide()
+
+	NamePlateDriverFrame:Hide()
+	NamePlateDriverFrame:UnregisterAllEvents()
+	NamePlateDriverFrame:RegisterEvent('NAME_PLATE_CREATED')
+	NamePlateDriverFrame:RegisterEvent('NAME_PLATE_UNIT_ADDED')
+	NamePlateDriverFrame:RegisterEvent('NAME_PLATE_UNIT_REMOVED')
+	NamePlateDriverFrame:HookScript('OnEvent', function(_, event, unit)
+		if(event == 'NAME_PLATE_UNIT_ADDED') then
+			if(not unit) then return end
+
+			local nameplate = C_NamePlate.GetNamePlateForUnit(unit)
+
+			if(nameplate.UnitFrame) then
+				nameplate.UnitFrame:SetParent(hiddenParent)
+			end
+		end
+	end)
+
+	local eventHandler = CreateFrame('Frame')
+	eventHandler:RegisterEvent('NAME_PLATE_UNIT_ADDED')
+	eventHandler:RegisterEvent('NAME_PLATE_UNIT_REMOVED')
+	eventHandler:RegisterEvent('PLAYER_TARGET_CHANGED')
+	eventHandler:RegisterEvent('PLAYER_LOGIN')
+	eventHandler:SetScript('OnEvent', function(_, event, unit)
+		if(event == 'PLAYER_LOGIN') then
+			if(not nameplateCVars) then return end
+
+			for cvar, value in next, nameplateCVars do
+				SetCVar(cvar, value)
+			end
+		elseif(event == 'PLAYER_TARGET_CHANGED') then
+			local nameplate = C_NamePlate.GetNamePlateForUnit('target')
+			if(not nameplate) then return end
+
+			nameplate.unitFrame:UpdateAllElements(event)
+
+			if(nameplateCallback) then
+				nameplateCallback(event, nameplate)
+			end
+		elseif(event == 'NAME_PLATE_UNIT_ADDED') then
+			if(not unit) then return end
+
+			unit = unit:lower()
+
+			local nameplate = C_NamePlate.GetNamePlateForUnit(unit)
+			if(not nameplate) then return end
+
+			nameplate.style = style
+
+			if(not nameplate.unitFrame) then
+				nameplate.unitFrame = CreateFrame('Button', prefix..nameplate:GetName(), nameplate)
+				nameplate.unitFrame:EnableMouse(false)
+
+				Private.UpdateUnits(nameplate.unitFrame, unit)
+
+				walkObject(nameplate.unitFrame, unit)
+			end
+
+			nameplate.unitFrame:SetAttribute('unit', unit)
+			nameplate.unitFrame:UpdateAllElements(event)
+
+			if(nameplateCallback) then
+				nameplateCallback(event, nameplate, unit)
+			end
+		elseif(event == 'NAME_PLATE_UNIT_REMOVED') then
+			if(not unit) then return end
```

Comment:
This (and all the other unit checks above) can be merged into the previous condition check.